File: android/app/src/main/kotlin/com/amazonaws/services/chime/flutterdemo/VideoTileView.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */
 
package com.amazonaws.services.chime.flutterdemo

import android.content.Context
import android.view.View
import io.flutter.plugin.platform.PlatformView
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.DefaultVideoRenderView
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoScalingType
import com.amazonaws.services.chime.sdk.meetings.utils.logger.ConsoleLogger

internal class VideoTileView(context: Context?, creationParams: Int?) : PlatformView {
    private val view: DefaultVideoRenderView

    private val videoTileViewLogger: ConsoleLogger = ConsoleLogger()

    override fun getView(): View {
        return view
    }

    override fun dispose() {}

    init {
        view = DefaultVideoRenderView(context as Context)
        view.scalingType = VideoScalingType.AspectFit
        MeetingSessionManager.meetingSession?.audioVideo?.bindVideoView(view, creationParams as Int)
            ?: videoTileViewLogger.error("VideoTileView", "Error while binding video view.")
    }
}

File: ios/Runner/VideoTileView.swift
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */
 
import AmazonChimeSDK
import Foundation

class VideoTileView: NSObject, FlutterPlatformView {
    private var _view: UIView

    init(
        frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) {
        _view = DefaultVideoRenderView()
        super.init()
           
        // Receieve tileId as a param.
        let tileId = args as! Int
        let videoRenderView = _view as! VideoRenderView
           
        // Bind view to VideoView
        MeetingSession.shared.meetingSession?.audioVideo.bindVideoView(videoView: videoRenderView, tileId: tileId)
           
        // Fix aspect ratio
        _view.contentMode = .scaleAspectFit
           
        // Declare _view as UIView for Flutter interpretation
        _view = _view as UIView
    }

    func view() -> UIView {
        return _view
    }
}
